sycl/test/regression/atomic_deprecated.cpp

```diff
@@ -0,0 +1,10 @@
+// RUN: %clangxx %fsycl-host-only -fsyntax-only -Xclang -verify -Xclang -verify-ignore-unexpected=note %s
+
+#include <sycl/sycl.hpp>
+int main() {
+  cl::sycl::multi_ptr<int, cl::sycl::access::address_space::global_space> a(
+      nullptr);
+  // expected-warning@+1 {{'atomic<int, sycl::access::address_space::global_space>' is deprecated: sycl::atomic is deprecated since SYCL 2020}}
```

Comment:
I would prefer to have this code in https://github.com/intel/llvm/blob/sycl/sycl/test/warnings/sycl_2020_deprecations.cpp where most of such deprecations are checked